### DISCUSSION TITLE
Fully download all required information prior to build?

### DISCUSSION THREAD

**Discussion by EmRowlands** [npm/feedback#355] [General]
I am trying to package an node-based project so that it can be built in a network sandbox using Gentoo's Portage. I have tried pre-running `npm install` and providing that for download, however, `npm run build` still connects to the registry during `fetchPackageMetaData`. Running a registry mirror/cache isn't really an option since the cache would still need to connect to upstream at some point during the build.

Is there any way to prevent this from happening or pre-download the metadata files? If not, is this a feasible feature?

**Comment by ljharb** [comment]
Unless you’re referring to a dependency’s install script, `npm run build` is running a “build” script defined in your project’s package.json, so what it does is entirely out of npm’s control.

**Reply by EmRowlands** [reply]
I shall have to investigate this further, thanks for the info.
